### DISCUSSION TITLE
Semgrep scans

### DISCUSSION THREAD

**Discussion by harshkolhatkar** [MrWong99/gitscanner#58] [General]
Hi, what's the expectation in performing static code analysis? Does it need to be kicked off on PRs/merges or on nightly basis? 
Do we have any infrastructure in place to run it offline?

**Comment by MrWong99** [comment]
Hi @harshkolhatkar, are you talking about #35?
This issue is meant to be performed on the client that runs gitscanner against the repositories that are supposed to be scanned. So they are run after the click in the UI.
Static code analysis for our repo itself is done via a Github action already upon each PR or release.

**Reply by harshkolhatkar** [reply]
Cool. So I'm thinking of exposing an API endpoint to call static code analysis with needed parameters and leveraging semgrep docker container to do the scanning in the backend. What'd you think?

**Reply by MrWong99** [reply]
Sounds good. I would prefer if semgrep would run in the same gitscanner process (to not need the dependency on Docker) but due to the nature of requiring different tools for static code analysis this is fine.
Looking forward to more input, this feature would be great to have! Should I assign the issue to you aswell?
